core/tx_pool.go

```diff
@@ -496,6 +496,42 @@ func (pool *TxPool) stats() (int, int) {
 	return pending, queued
 }
 
+// StatsWithMinBaseFee retrieves the current pool stats, namely the number of pending and the
+// number of queued (non-executable) transactions greater equal minBaseFee.
+func (pool *TxPool) StatsWithMinBaseFee(minBaseFee *big.Int) (int, int) {
+	pool.mu.RLock()
+	defer pool.mu.RUnlock()
+
+	return pool.statsWithMinBaseFee(minBaseFee)
+}
+
+// statsWithMinBaseFee retrieves the current pool stats, namely the number of pending and the
+// number of queued (non-executable) transactions greater equal minBaseFee.
+func (pool *TxPool) statsWithMinBaseFee(minBaseFee *big.Int) (int, int) {
+	pending := 0
+	for _, list := range pool.pending {
+		txs := list.Flatten()
```

Comment:
`Flatten` causes copy, any chance we can avoid the call to `Flatten`?

Reply:
You're right we don't need the copy here since we only read. If your only concern is with the copy, we could just call `flatten()` (lowercase) which does not copy but returns the cache directly. 

https://github.com/scroll-tech/go-ethereum/blob/bfe803001be452e90e4f9208d8a1cc24e5c44ef9/core/tx_list.go#L222

Reply:
Changed in https://github.com/scroll-tech/go-ethereum/pull/884/commits/4fbc45cf538e0239fbfd8735d2ea3dc7ed661b3d

Reply:
flatten might still cause a copy if the cache is not populated but I guess it is not worth trying to get rid of it since you want to iterate over the txns in a sorted way.